### PR TITLE
Small improvements for the Nim program

### PR DESCRIPTION
**Pull request by alaviss**

### CODE REVIEW

**Comment by frol** [comment]
@def- @PMunch @Yardanico @narimiran Could you help to review this change in terms of how "naive" Nim implementation will be with the change?

**Comment by PMunch** [comment]
No magic there, just slightly more idiomatic code as per the Nim style-guidelines. That is probably what it would have ended up looking like if I wrote it myself from scratch.

File: nim/main.nim [hunk 1]
```diff
@@ -5,67 +5,64 @@ type Node = ref object
   left, right: Node
 
 proc newNode(x: int32): Node =
-  return Node(x: x, y: rand(high(int32).int).int32, left: nil, right: nil)
+  result = Node(x: x, y: rand(high int32).int32)
```
Comment (Yardanico) [inline nim/main.nim]: This is very often used in Nim, and this is the preferred way most of the time.

**Comment by frol** [comment]
@alaviss Thanks for your contribution!

**Pull request merged**